domain/externalcontroller/state/state.go

```diff
@@ -59,125 +65,235 @@ WHERE  	ctrl.uuid = $M.id`
 		return nil, errors.NotFoundf("external controller %q", controllerUUID)
 	}
 
-	return controllerInfoFromRows(controllerUUID, rows), nil
+	return controllerInfoFromRows(rows), nil
 }
 
-func (st *State) ControllerForModel(
-	ctx context.Context,
-	modelUUID string,
-) (*crossmodel.ControllerInfo, error) {
+func (st *State) ControllersForModels(ctx context.Context, modelUUIDs ...string) ([]crossmodel.ControllerInfo, error) {
 	db, err := st.DB()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
+	// TODO(nvinuesa): We should use an `IN` statement query here, instead
+	// of looping over the list of models and performing N queries, but
+	// they are not yet supported on sqlair:
+	// https://github.com/canonical/sqlair/pull/76
 	q := `
-SELECT 	(ctrl.uuid, alias, ca_cert, address) as &ExternalController.* 
-FROM   	external_controller AS ctrl	
-	JOIN external_model AS model
-	ON ctrl.uuid = model.controller_uuid
-       	LEFT JOIN external_controller_address AS addrs
-       	ON ctrl.uuid = addrs.controller_uuid
-WHERE  	model.uuid = $M.id`
+SELECT (ctrl.uuid,  
+       ctrl.alias,
+       ctrl.ca_cert,
+       addrs.address) as &ExternalController.*,
+       model.uuid as &ExternalController.model
+FROM external_controller AS ctrl	
```

Comment:
Right align these lines like in the query you're replacing.

Reply:
done, see 95564b14e4c4efb7653904e05c7eeab5ab1bbe9b